rust/frontend/src/function.rs

```diff
@@ -18,21 +18,8 @@ use std::{
 use crate::{ts, ErrorKind, Module, Result, TVMArgValue, TVMRetValue, TVMTypeCode, TVMValue};
 
 lazy_static! {
-    static ref GLOBAL_FUNCTIONS: Mutex<BTreeMap<&'static str, Option<Function>>> = {
-        let mut out_size = 0 as c_int;
-        let name = ptr::null_mut() as *mut c_char;
-        let mut out_array = name as *mut _;
-        check_call!(ts::TVMFuncListGlobalNames(
-            &mut out_size as *mut _,
-            &mut out_array
-        ));
-        let names_list = unsafe { slice::from_raw_parts(out_array, out_size as usize) };
-        Mutex::new(
-            names_list
-                .into_iter()
-                .map(|&p| (unsafe { CStr::from_ptr(p).to_str().unwrap() }, None))
-                .collect(),
-        )
+    static ref GLOBAL_FUNCTIONS: Mutex<HashMap<String, Function>> = {
+        Mutex::new(HashMap::new())
```

Comment:
I believe `TVMFuncRegisterGlobal` is meant for adding new global functions at runtime, which means that the set of names may change?

Reply:
Yes, they could change, but functions can also be removed which makes caching _at all_ ill advised. The de-facto semantics of TVM globals is static registration. Always re-wrapping pointers might be an option but I'm not sure if the performance cost would be worth the (unused) additional flexibility.

On the bright side, part of the reason for the rust runtime has been to remove these pesky globals. As we further expand the rust functionality, we won't have to think as much about C++ interop.

Once I finish up unifying the tvm crate and merging in your PR, a good next area of focus might be on creating device apis to enable use cases like GPUs (let's start with AMD because you seem to have a real use case) and VTA.